tests/test_pointing.py

```diff
@@ -0,0 +1,275 @@
+"Tests for pointing.py"
+
+# These tests only use 1 polarization axis
+import numpy as np
+import katpoint
+import pytest
+from katsdpcalproc import pointing
+from katpoint import (rad2deg, lightspeed)
+
+# Creating metadata fpr tests
+middle_time = 1691795333.43713
+temperature = 14.7
+humidity = 29.1
+pressure = 897
+track_duration = 24
+NUM_CHUNKS = 5
+pols = ["h"]
+
+centre_freq = 1284000000.0
+bandwidth = 856000000.0
+no_channels = 1000
+
+# Calculating channel and chunk freqencies
+channel_freqs = centre_freq + (np.arange(no_channels) - no_channels / 2) * (bandwidth / no_channels)
+chunk_freqs = channel_freqs.reshape(NUM_CHUNKS, -1).mean(axis=1)
+target = katpoint.Target(
+    body="J1939-6342, radec bfcal single_accumulation, 19:39:25.03, -63:42:45.6")
+
+# Maximum distance of offset from target, in degrees
+max_extent = 1.0
+num_pointings = 8
+# Build up sequence of pointing offsets running linearly in x and y directions
+scan = np.linspace(-max_extent, max_extent, num_pointings // 2)
+offsets_along_x = np.c_[scan, np.zeros_like(scan)]
+offsets_along_y = np.c_[np.zeros_like(scan), scan]
+offsets = np.r_[offsets_along_y, offsets_along_x]
+
+# Creating list of antenna objects
+ants = [
+    katpoint.Antenna(
+        'm000',
+        '-30:42:39.8',
+        '21:26:38.0',
+        1086.6,
+        diameter=15,
+        beamwidth=1.22,
+        pointing_model="""0:04:20.6,0,0:01:14.2,0:02:58.5,0:00:05.1,0:00:00.4,
+                          0:20:04.1,-0:00:34.5,0,0,-0:03:10.0,0,0,0,0,0,0,0,0,0,0,0""",
+        delay_model="-8.264,-207,8.6,212.6,212.6,1"),
+    katpoint.Antenna(
+        'm001',
+        '-30:42:39.8',
+        '21:26:38.0',
+        1086.6,
+        diameter=15,
+        beamwidth=1.22,
+        pointing_model="""0:04:15.6,0,0:01:09.2,0:01:58.5,0:00:05.1,0:00:00.4,
+                          0:16:04.1,-0:00:34.5,0,0,-0:03:10.0,0,0,0,0,0,0,0,0,0,0,0""",
+        delay_model="-8.264,-207,8.6,212.6,212.6,1")]
+
+az_el_adjust = np.zeros((len(ants), 2))
+
+# Creating gains, numpy array shape (no.offsets, no.polarisations, no.antennas)
+weights = np.ones(10)
+just_gains = []
+for i in range(0, len(offsets)):
+    gg = []
+    for j in pols:
+        gg.append(np.array(np.ones(len(ants))))
+    just_gains.append(np.array(gg))
+just_gains = np.array(just_gains)
+
+
+# Creating bp_gains, numpy array of shape (no.offsets, no.freqs, no.polarisations, no.antennas)
+
+def g_o_g(offsets, ants, channel_freqs):
+    bp_gains3 = []
+    for i in range(0, len(offsets)):
+        bp_gains2 = []
+        for f in range(0, len(channel_freqs)):
+            bp_gains1 = []
+            for pol in pols:
+                ex_width = []
+                for a, ant in enumerate(ants):
+                    # expected widths for each frequency channel
+                    expected_width = rad2deg(ant.beamwidth * lightspeed
```

Comment:
This snippet is a copy and paste of the part in `beam_fit`. It probably needs to go into a separate helper function in `pointing.py` to avoid breaking the tests if you ever adjust this.